docs/notifications/index.md

```diff
@@ -218,6 +218,59 @@ React.useEffect(() => {
 }, [lastSignal, notificationsApi]);
 ```
 
+#### Using signals in your own plugin
+
+It's possible to use signals in your own plugin to deliver data from the backend to the frontend in near real-time.
+
+To use signals in your own frontend plugin, you need to add the `useSignal` hook from `@backstage/plugin-signals-react` from `@backstage/plugin-notifications-common` with optional generic type of the signal.
+
+```ts
+// To use the same type of signal in the backend, this should be placed in a shared common package
+export type MySignalType = {
+  user: string;
+  data: string;
+  // ....
+};
+
+const { lastSignal } = useSignal<MySignalType>('my-plugin');
```

Comment:
Again, I'm sorry but I think that `useSignal` may not be a great first interface to subject people to, for the reasons discussed in the other PR about reactivity. Leaning on `useObservable` instead or similar will be likely to be the superior interface, imo. Of course we'll accept this as the `useSignal` hook could be one of several things in the toolbox either way, and this documents the current state of things which is great. Just noting though for the future

Reply:
Agree, there's the other PR that enables the possibility of observing signals #25959. If we decide to deprecate this hook, it will require some time for plugins using the signals to adapt. Anyhow, I think making it observable makes much more sense than having this hook.